src/test/java/org/jenkinsci/plugins/p4/scm/PerforceScmSourceTest.java

```diff
@@ -33,16 +44,159 @@ public void testMultiBranchWithStreams() throws Exception {
 		String credential = auth.getId();
 
 		String format = "jenkins-${NODE_NAME}-${JOB_NAME}";
-		String includes = "//streams/...";
+		String includes = "//stream/...";
 		SCMSource source = new StreamsScmSource("streams", credential, includes, null, format, null);
 
 		WorkflowMultiBranchProject multi = jenkins.jenkins.createProject(WorkflowMultiBranchProject.class, "multi-streams");
-		multi.getSCMSourceCriteria(source);
+		multi.getSourcesList().add(new BranchSource(source));
+		multi.scheduleBuild2(0);
+		jenkins.waitUntilNoActivity();
+
+		assertThat("We now have branches",
+				multi.getItems(), not(is((Collection<WorkflowJob>) Collections.<WorkflowJob>emptyList())));
+
+		WorkflowJob job = multi.getItem("Ace-main");
+
+		assertThat("We now have a branch", job, notNullValue());
+
+		WorkflowRun build = job.getLastBuild();
+
+		assertThat("The branch was built", build, notNullValue());
+		assertThat("The branch was built", build.getNumber(), is(1));
+	}
+
+	@Test
+	public void testMultiBranchWithClassic() throws Exception {
+
+		String credential = auth.getId();
+
+		String format = "jenkins-${NODE_NAME}-${JOB_NAME}";
+		String includes = "//stream/...";
+		SCMSource source = new BranchesScmSource("classic", credential, includes, null, format, null);
+
+		WorkflowMultiBranchProject multi = jenkins.jenkins.createProject(WorkflowMultiBranchProject.class, "multi-classic");
+		multi.getSourcesList().add(new BranchSource(source));
+		multi.scheduleBuild2(0);
+		jenkins.waitUntilNoActivity();
+
+		assertThat("We now have branches",
+				multi.getItems(), not(is((Collection<WorkflowJob>) Collections.<WorkflowJob>emptyList())));
```

Comment:
`not(containsInAnyOrder())` usually is simpler and requires less generics mangling (and yep not exactly obvious but it verifies that you do not have an empty collection)